File: src/main/java/com/utsav/ems/service/EmployeeService.java
package com.utsav.ems.service;

import java.util.List;

import com.utsav.ems.model_bean.Employee;

public interface EmployeeService {
	Employee saveEmployee(Employee employee);
	List<Employee> getAllEmployees();
	Employee getEmployeeById(long id);
	Employee updateEmployee(Employee employee, long id);
	void deleteEmployee(long id);
}

File: ReactApp/src/Service/EmployeeService.js
import axios from 'axios'
const EMPLOYEE_BASE_REST_API_URL = 'http://localhost:9000/api/employees';

class EmployeeService{

    getAllEmployees(){
        return axios.get(EMPLOYEE_BASE_REST_API_URL)
    }

    createEmployee(employee){
        return axios.post(EMPLOYEE_BASE_REST_API_URL, employee)
    }

    getEmployeeById(employeeId){
        return axios.get(EMPLOYEE_BASE_REST_API_URL + "/" + employeeId)
    }

    updateEmployee(employeeId, employee){
        return axios.put(EMPLOYEE_BASE_REST_API_URL + "/" + employeeId, employee)
    }

    deleteEmployee(employeeId, employee){
        return axios.delete(EMPLOYEE_BASE_REST_API_URL + "/" + employeeId)
    }
}

export default new EmployeeService()
